app/assets/javascripts/govuk_publishing_components/analytics-ga4/ga4-link-tracker.js

```diff
@@ -52,6 +52,10 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
 
         for (var i = 0; i < classes.length; i++) {
           if (target.closest('.' + classes[i].trim())) {
+            // Stops the link tracker firing twice if the link itself has data-ga4-link, and the parent element does as well.
```

Comment:
I see you've done quite a bit of work to fix this, but can you explain how this problem arose? Do we have govspeak components that are already wrapped with the link tracker? If that's the case we should ideally remove that tracking, if the component will now track by default. Would that remove the need for this extra code?

Reply:
Oh I see. Okay, that's fair enough. However it does make me wonder if there would be a cleaner way of doing tracking in govspeak, now we know what the complexities are (outside of the scope of this PR). Maybe something to consider separately.

Reply:
@andysellick sounds good 👍